### ISSUE TITLE
Make "arrayjoin: not one band or 4 bands" error report more helpful

### ISSUE THREAD

**Issue by int-ua** [libvips/pyvips#202]
**Is your feature request related to a problem? Please describe.**
When getting this error on a script someone else wrote it's not obvious what's happening and what the problem is without searching an explanation on the web.

**Describe the solution you'd like**
Adding something like 

>You may need to convert all images to the same color space.
> `pyvips.Image.colourspace` function may help with that.

**Additional context**
Trying to just generate a text on a _transparent_ background is unnecessarily complex and lacks an example that's easy to find.

**Comment by jcupitt** [comment]
The main docs for arrayjoin have more detail:

https://libvips.github.io/libvips/API/current/libvips-conversion.html#vips-arrayjoin

You're right that they could mention `colourspace`.

**Comment by int-ua** [comment]
May I suggest also changing what user of an implementation that uses `libvips` will see? I guess it's WIP, but I've decided to ask explicitly.

**Comment by jcupitt** [comment]
Sorry, I'm not sure I understand. Could you explain a little more?

**Comment by int-ua** [comment]
Can the text that is shown _to the user_ when this error happens be expanded with the same info? Context: https://github.com/CleverRaven/Cataclysm-DDA/issues/43157

**Comment by int-ua** [comment]
It was completely not obvious what is happening as I didn't even write the script that showed me this error.

**Comment by int-ua** [comment]
I am now trying to fix the aforementioned `compose.py`. `colourspace` method docs are missing [a link to] the link of possible values for `space` parameter.

**Comment by jcupitt** [comment]
`Interpretation` is documented here:

https://libvips.github.io/pyvips/enums.html#pyvips.enums.Interpretation

**Comment by int-ua** [comment]
Yes, after a bit of searching I found it. What I'm saying is that a link to that place from the `colourspace` method would help a lot. Anyway. How am I supposed to load `b-w` images for `srgb` `arrayjoin` later if `colourspace` doesn't change the interpretations for some reason?

```py
                vips_image = Vips.Image.pngload(png_pathname)
                if vips_image.interpretation != 'srgb':
                    print(vips_image.interpretation)
                    vips_image.colourspace('srgb')
                    print(vips_image.interpretation)
```
`interpretation` stays `b-w` and the error persists with this code.

**Comment by int-ua** [comment]
ok, that was my error, sorry, figured it out.

**Comment by jcupitt** [comment]
I made an enhancement issue for linking enum args to the relevant enum, thanks for the suggestion.

**Comment by jcupitt** [comment]
Args should link to enums now, eg.:

https://libvips.github.io/pyvips/vimage.html#pyvips.Image.colourspace

I credited you in the changelog.
